services/garden/gardenSlice.ts

```diff
@@ -6,19 +6,33 @@ import { RootState } from "../../store";
 
 export const gardenSlice = createSlice({
   name: "gardens",
-  initialState: initialGardenState as Garden[],
+  initialState: initialGardenState as unknown as Garden[],
```

Comment:
The "unknown as" leaves the initialState untyped, you would have been getting a bunch of errors because of your changes to initialGardenState but the type wasn't changed to with those changes, but you wouldn't have seen any errors in initialGardenState because I didn't type annotate it and only annotated it within the reducer with the as Guarden[].

I'll just merge this PR and do a type fix PR, it's my fault because I've added types for fields we haven't implemented yet and haven't properly gone through what types are optional. I also should have annotated initialGuardenState